src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -127,6 +131,45 @@ public Date read(JsonReader jsonReader) throws IOException {
     }
 
     /**
+     * TypeAdaptorFactory creating TypeAdapters for Optional
+     */
+    private class OptionalTypeAdapterFactory implements TypeAdapterFactory {
+
+        @SuppressWarnings("unchecked")
+        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken) {
```

Comment:
create is overridden from `TypeAdapterFactory` so its already documented there i put the `@Override` on it to make it clear.